frontend/src/components/collaborator/CollaboratorEdit.vue

```diff
@@ -0,0 +1,199 @@
+import IconButton from '@/components/buttons/IconButton.vue'
+import SettingsForm from '@/components/dialog/SettingsForm.vue'
+import ButtonDelete from '@/components/buttons/ButtonDelete.vue'
+import DialogEntityDelete from '@/components/dialog/DialogEntityDelete.vue'
+import DialogCollaboratorDeactivate from '@/components/collaborator/DialogCollaboratorDeactivate.vue'
+import { errorToMultiLineToast } from '@/components/toast/toasts.js'
+import CollaboratorListItem from '@/components/collaborator/CollaboratorListItem.vue'
+
+export default {
+  name: 'CollaboratorEdit',
+  components: {
+    CollaboratorListItem,
+    DialogCollaboratorDeactivate,
+    DialogEntityDelete,
+    ButtonDelete,
+    SettingsForm,
+    IconButton,
+    SettingsCollaboratorForm,
+  },
+  extends: DialogBase,
+  mixins: [campRoleMixin],
+  props: {
+    collaborator: { type: Object, required: true },
+    disabled: { type: Boolean, default: false },
+    inactive: { type: Boolean, default: false },
+  },
+  data() {
+    return {
+      resendingEmail: false,
+      entityProperties: ['camp', 'inviteEmail', 'role', 'status'],
+      entityUri: '/camp_collaborations',
+    }
+  },
+  computed: {
+    camp() {
+      return this.collaborator.camp()
+    },
+    isLastManager() {
+      if (this.collaborator.status !== 'established') return false
+      if (this.collaborator.role !== 'manager') return false
+      const camp = this.collaborator.camp()
+      return (
+        camp
+          ?.campCollaborations()
+          ?.items?.filter((collaborator) => collaborator.status === 'established')
+          .filter((collaborator) => collaborator.role === 'manager').length <= 1
+      )
+    },
+    isOwnCampCollaboration() {
+      if (!(typeof this.collaborator.user === 'function')) {
+        return false
+      }
+      return this.$store.state.auth.user?.id === this.collaborator.user().id
+    },
+    name() {
+      return this.collaborator.user
+        ? this.collaborator.user().displayName
+        : this.collaborator.inviteEmail
+    },
+  },
+  watch: {
+    showDialog: function (showDialog) {
+      if (showDialog) {
+        this.entityUri = this.collaborator._meta.self
+        this.setEntityData({
+          role: this.collaborator.role,
+        })
+      } else {
+        // clear form on exit
+        this.clearEntityData()
+      }
+    },
+  },
+  methods: {
+    resendInvitation() {
+      this.resendingEmail = true
+      this.api
+        .href(this.api.get(), 'campCollaborations', {
+          id: this.collaborator.id,
+          action: 'resend_invitation',
+        })
+        .then((postUrl) => this.api.patch(postUrl, {}))
+        .catch((e) => this.$toast.error(errorToMultiLineToast(e)))
+        .finally(() => {
+          this.resendingEmail = false
+        })
+    },
+    reinvite() {
+      this.resendingEmail = true
+      this.api.patch(this.collaborator, { status: 'invited' }).then(() => {
```

Comment:
```suggestion
      this.api.patch(this.collaborator, { status: 'invited' }).finally(() => {
```
Else the animation does not stop when an error occurs